nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -185,6 +185,9 @@ impl DataStore {
         // join ip_pool to ip_pool_resource and filter
 
         // used in both success and error outcomes
+        // TODO: Correctness
```

Comment:
This is an issue @zephraph and I ran into while troubleshooting an error during instance creation. When this lookup is attempted and fails, the logs will show "Default pool for current silo" instead of the id used to attempt the lookup. I'm not quite sure how this is _supposed_ to work, but I think it's safe to say that the error message should at least indicate the name or uuid used to attempt the lookup.

Reply:
cc @david-crespo

Reply:
I think Ryan is correct that it's not a correctness issue. But it should be fixed, thanks for flagging it as a source of pain. I see #4864. I'm still working on IP pools stuff this week so I'll roll in an improvement.

Reply:
@david-crespo After reading the comment you left in #4864 I understand things better now, many thanks!